tools/clang/lib/Sema/SemaHLSL.cpp

```diff
@@ -9695,6 +9694,13 @@ bool HLSLExternalSource::ValidateCast(
 
     if (!suppressErrors)
     {
+      bool IsOutputParameter = false;
+      if (clang::DeclRefExpr *OutFrom = dyn_cast<clang::DeclRefExpr>(sourceExpr)) {
+        if (ParmVarDecl *Param = dyn_cast<ParmVarDecl>(OutFrom->getDecl())) {
+          IsOutputParameter = Param->isModifierOut();
+        }
+      }
+
```

Comment:
This was incidental. At one point I was trying to emit the warning here and discovered how to solve this problem though I couldn't put the other warning here.